tests/multimodel/test_adex.py

```diff
@@ -0,0 +1,316 @@
+            self.assertTrue(
+                all(result[state_var].shape == (int(DURATION / DT), 1) for state_var in adex.state_variable_names[0])
+            )
+            all_results.append(result)
+        # test results are the same from different backends
+        for state_var in all_results[0]:
+            corr_mat = np.corrcoef(
+                np.vstack([result[state_var].values.flatten().astype(float) for result in all_results])
+            )
+            self.assertTrue(np.greater(corr_mat, CORR_THRESHOLD).all())
+
+    def test_compare_w_neurolib_native_model(self):
+        """
+        Compare with neurolib's native ALN model.
+        """
+        # run this model
+        adex_multi = self._create_node()
+        multi_result = adex_multi.run(DURATION, DT, ZeroInput(DURATION, DT).as_array(), backend="numba")
+        # run neurolib's model
+        aln_neurolib = ALNModel(seed=SEED)
+        aln_neurolib.params["duration"] = DURATION
+        aln_neurolib.params["dt"] = DT
+        aln_neurolib.run()
+        for (var_multi, var_neurolib) in NEUROLIB_VARIABLES_TO_TEST:
+            corr_mat = np.corrcoef(aln_neurolib[var_neurolib], multi_result[var_multi].values.T)
+            self.assertTrue(np.greater(corr_mat, CORR_THRESHOLD).all())
+
+
+class TestAdExNetwork(unittest.TestCase):
+    SC = np.random.rand(2, 2)
+    np.fill_diagonal(SC, 0.0)
+    DELAYS = np.array([[0.0, 7.8], [7.8, 0.0]])
+
+    def test_init(self):
+        adex = AdExNetwork(self.SC, self.DELAYS)
+        self.assertTrue(isinstance(adex, AdExNetwork))
+        self.assertEqual(len(adex), self.SC.shape[0])
+        self.assertEqual(adex.initial_state.shape[0], adex.num_state_variables)
+        self.assertEqual(adex.default_output, f"q_mean_{EXC}")
+
+    def test_run(self):
+        adex = AdExNetwork(self.SC, self.DELAYS, exc_seed=SEED, inh_seed=SEED)
+        all_results = []
+        for backend, noise_func in BACKENDS_TO_TEST.items():
+            result = adex.run(
+                DURATION, DT, noise_func(ZeroInput(DURATION, DT, adex.num_noise_variables)), backend=backend,
+            )
+            self.assertTrue(isinstance(result, xr.Dataset))
+            self.assertEqual(len(result), adex.num_state_variables / adex.num_nodes)
+            self.assertTrue(all(result[result_].shape == (int(DURATION / DT), adex.num_nodes) for result_ in result))
+            all_results.append(result)
+        # test results are the same from different backends
+        for state_var in all_results[0]:
+            all_ts = np.vstack([result[state_var].values.flatten().astype(float) for result in all_results])
+            if np.isnan(all_ts).any():
+                continue
+            corr_mat = np.corrcoef(all_ts)
+            print(corr_mat)
+            self.assertTrue(np.greater(corr_mat, CORR_THRESHOLD).all())
+
+    def test_compare_w_neurolib_native_model(self):
```

Comment:
@caglarcakan here is the comparison between native ALN and this one for a small dummy network